### ISSUE TITLE
[Compilation][Windows][C++17] Error during compilation

### ISSUE THREAD

**Issue by loumalouomega** [KratosMultiphysics/Kratos#10137]
**Description**

~~~
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\include\xutility(1169): warning C4996: 'std::iterator<IC,T,ptrdiff_t,T *,T &>::iterator_category': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
~~~

**Scope**

- Compilation
- KratosCore

**To Reproduce**

Windows 10. MSVSC 2019.

**Expected behavior**
To compile

**Environment**

- OS: Windows 10
- Branch: master
- MSVSC 2019.

**Comment by loumalouomega** [comment]
(yes, the proposed flags solve the issue, I just wanted to notify)

**Comment by roigcarlo** [comment]
A little more context? xutility is pretty general, which file/s trigger this?

**Comment by loumalouomega** [comment]
The message is long. It is the std::iterator derivation

**Comment by loumalouomega** [comment]
> A little more context? xutility is pretty general, which file/s trigger this?

The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators.

**Comment by loumalouomega** [comment]
Now for some reason this error doesn't appear anymore :P

**Comment by loumalouomega** [comment]
It's happening again

**Comment by loumalouomega** [comment]
> It's happening again

Fixed with #10270
